R/agyw-integration.R

```diff
@@ -0,0 +1,1659 @@
+                  `new40-44all`,`new40-44f`,`new40-44m`,
+                  `Inci40-44f`,`Incicategory40-44f`,`Inci40-44m`,`Incicategory40-44m`,
+                  `Pop45-49all`,`Pop45-49f`,`Pop45-49m`,
+                  `PLHIV45-49all`,`PLHIV45-49f`,`PLHIV45-49m`,
+                  `new45-49all`,`new45-49f`,`new45-49m`,
+                  `Inci45-49f`,`Incicategory45-49f`,`Inci45-49m`,`Incicategory45-49m`,
+                  `Pop15-49all`,`Pop15-49f`,`Pop15-49m`,
+                  `PLHIV15-49all`,`PLHIV15-49f`,`PLHIV15-49m`,
+                  `new15-49all`,`new15-49f`,`new15-49m`,
+                  `Inci15-49f`,`Incicategory15-49f`,`Inci15-49m`,`Incicategory15-49m`)
+
+  ## Clean up area names and Country names
+  if(options$area_scope=="AGO") {
+    naomi_wide$area_name <- stringr::str_to_title(naomi_wide$area_name)
+  }
+
+  if(options$area_scope %in% c("TCD","GIN")) {
+    naomi_wide$area_name <- iconv(naomi_wide$area_name, from="UTF-8",to="LATIN1")
+  }
+
+  v <- list(naomi_long = df2,
+            naomi_wide = naomi_wide)
+
+  }
+
+
+
+#' Dissagreggate admin1 FSW proportions from Oli's KP model to 5-age groups
+#'
+#' @param outputs Naomi output.
+#' @param options Naomi model options.
+#' @param naomi_pop Naomi population estimates for T2.
+#' @param kp_consensus Key pop consensus estimates.
+#'
+#'
+#' @return District level FSW estimates by 5-year age bands for ages 15-49.
+#' @keywords internal
+
+
+agyw_disaggregate_fsw <- function(outputs,
+                                  options,
+                                  naomi_pop,
+                                  kp_consensus){
+
+  # Extract country specific national FSW PSEs
+  iso3 <- options$area_scope
+
+  pse <- naomi.resources::load_agyw_exdata("kp_estimates", iso3) %>%
+    dplyr::filter(kp == "FSW", indicator == "pse_prop")
+
+  fsw_pse <- pse %>%
+    dplyr::rename(prop_fsw = median) %>%
+    dplyr::select(-indicator,-lower,-upper)
+
+  age_groups <- c("Y015_019", "Y020_024", "Y025_029", "Y030_034",
+                  "Y035_039", "Y040_044", "Y045_049")
+
+  # Calculating FSW proportion of total female population
+  fsw <- fsw_pse %>%
+    dplyr::mutate(age_group = "Y015_049") %>%
+    dplyr::left_join(naomi_pop %>% dplyr::filter(sex == "female"),
+                     by = dplyr::join_by(iso3, area_id, age_group)) %>%
+    dplyr::mutate(total_fsw = population * prop_fsw) %>%
+    dplyr::select(iso3, area_id, total_fsw, age_group, area_level, spectrum_region_code)
+
+  # Check for consensus estimate of FSW
+  fsw_consensus <- kp_consensus[kp_consensus$key_population == "FSW", ]$population_size
+
+  if(!is.na(fsw_consensus)){
+
+    # Check if consensus estimate is larger than age matched population denominator
+    pop <- naomi_pop[naomi_pop$area_level == 0 & naomi_pop$age_group == "Y015_049" & naomi_pop$sex == "female",]$population
+    prop_pop <- fsw_consensus / pop
+
+    if(prop_pop >= 0.05) {
+      t_("FSW_CONSENSUS_WARNING")
```

Comment:
This call doesn't go anywhere? Did you mean it as a waning

Reply:
Have removed and fixed up test to check that this working.